opentelemetry-sdk/src/resource/mod.rs

```diff
@@ -261,6 +269,65 @@ pub trait ResourceDetector {
     fn detect(&self) -> Resource;
 }
 
+/// Builder for [Resource]
+#[derive(Debug)]
+pub struct ResourceBuilder {
+    resource: Resource,
+}
+
+impl ResourceBuilder {
+    /// Add a single [ResourceDetector] to your resource.
+    pub fn with_detector(self, detector: Box<dyn ResourceDetector>) -> Self {
+        self.with_detectors(&[detector])
+    }
+
+    /// Add multiple [ResourceDetector] to your resource.
+    pub fn with_detectors(mut self, detectors: &[Box<dyn ResourceDetector>]) -> Self {
+        self.resource = self.resource.merge(&Resource::from_detectors(detectors));
+        self
+    }
+
+    /// Add a [KeyValue] to the resource.
+    pub fn with_attribute(self, kv: KeyValue) -> Self {
+        self.with_attributes(vec![kv])
```

Comment:
We can avoid this `vec!` allocation and use any array instead.

```suggestion
        self.with_attributes([kv])
```

Reply:
updated, and updated the migration guide